Library/Homebrew/livecheck/strategy/pypi.rb

```diff
@@ -20,7 +20,7 @@ class Pypi
         NICE_NAME = "PyPI"
 
         # The `Regexp` used to determine if the strategy applies to the URL.
-        URL_MATCH_REGEX = /files\.pythonhosted\.org/i.freeze
+        URL_MATCH_REGEX = %r{^https?://files\.pythonhosted\.org/packages(?:/[^/]+){4}i}.freeze
```

Comment:
Ensures that the path has exactly 4 parts because 4th is the filename used for package name extraction.

```rb
`https://files.pythonhosted.org/packages/<hex>/<hex>/<long_hex>/example-1.2.3.tar.gz`
```